lib/Core/TerriaError.ts

```diff
@@ -53,69 +99,100 @@ export type TerriaErrorOverrides = Partial<TerriaErrorOptions> | string;
 export default class TerriaError {
   private readonly _message: string | I18nTranslateString;
   private readonly _title: string | I18nTranslateString;
-  private readonly useTerriaErrorNotification: boolean;
+  private _raisedToUser: boolean = false;
 
+  readonly severity: TerriaErrorSeverity | (() => TerriaErrorSeverity);
   /** `sender` isn't really used for anything at the moment... */
   readonly sender: unknown;
   readonly originalError?: (TerriaError | Error)[];
-  private _raisedToUser: boolean = false;
+
+  @observable showDetails = false;
 
   /**
    * Convenience function to generate a TerriaError from some unknown error. It will try to extract a meaningful message from whatever object it is given.
    *
-   * If error is a `TerriaError`, then `createParentError` it will be used to create a tree of `TerriaErrors` (see {@link `TerriaError#createParentError}`).
-   *
    * `overrides` can be used to add more context to the TerriaError
+   *
+   * If error is a `TerriaError`, and `overrides` are provided -  then `createParentError` will be used to create a tree of `TerriaErrors` (see {@link `TerriaError#createParentError}`).
+   *
+   * Note, you can not pass `TerriaErrorOptions` (or JSON version of `TerriaError`) as the error parameter.
+   *
+   * For example:
+   *
+   * This is  **incorrect**:
+   *
+   * ```
+   * TerriaError.from({message: "Some message", title: "Some title"})
+   * ```
+   *
+   * Instead you must use TerriaError constructor
+   *
+   * This is **correct**:
+   *
+   * ```
+   * new TerriaError({message: "Some message", title: "Some title"})
+   * ```
    */
   static from(error: unknown, overrides?: TerriaErrorOverrides): TerriaError {
     if (error instanceof TerriaError) {
       return isDefined(overrides) ? error.createParentError(overrides) : error;
     }
 
-    // Try to find message using overrides or error object
-    let message = "Unknown error";
+    // Try to find message from error object
+    let message: string | undefined;
+    // Create original Error from `error` object
+    let originalError: Error | undefined;
 
-    // If overrides is a string - we treat is as the `message` parameter
-    if (typeof overrides === "string") {
-      message = overrides;
-      overrides = {};
-    } else if (typeof error === "string") {
+    if (typeof error === "string") {
       message = error;
+      originalError = new Error(message);
     } else if (error instanceof Error) {
       message = error.message;
+      originalError = error;
     } else if (typeof error === "object" && error !== null) {
       message = error.toString();
```

Comment:
this feels a bit strange and we might get a lot of "[object Object]" in our error reporting :/ I think we should only allow Error types and force the caller to instatiate their own Errors

Reply:
I disagree, the point of `TerriaError.from()` is that is accepts `error:unkown`:
https://github.com/TerriaJS/terriajs/blob/54cf1fc85f38d425274f233a77bd21bc6e3e4dbe/lib/Core/TerriaError.ts#L136

I think its quite unlikely that a random object will get thrown up the chain. 

I'm happy to replace `"[object Object]"` with `"Unknown Error"` or something.

This approach covers Cesium's error objects - for example `DeveloperError`

https://github.com/TerriaJS/cesium/blob/terriajs/Source/Core/DeveloperError.js#L57-L65